lib/ansible/module_utils/redfish_utils.py

```diff
@@ -505,16 +505,21 @@ def manage_indicator_led(self, command):
         return result
 
     def manage_system_power(self, command):
-        result = {}
         key = "Actions"
 
         # Search for 'key' entry and extract URI from it
         response = self.get_request(self.root_uri + self.systems_uris[0])
         if response['ret'] is False:
             return response
-        result['ret'] = True
         data = response['data']
         power_state = data["PowerState"]
+
+        if power_state == "On" and command == 'PowerOn':
+            return {'ret': True, 'changed': False}
```

Comment:
should we also add support to "ForceOn", "ForceRestart", "PushPowerButton", "Nmi", and "PowerCycle" that are supported in the Redfish ComputerSystem ResetType enum ?

Reply:
We certainly can add those commands if you think they are needed/valuable. If so, please open a new issue to add that support. (For the current issue/PR I want to fix the existing commands.)

Reply:
I agree with ForceOn, but the other values I think would have some immediate impact.

- ForceRestart I wouldn't expect to affect the power state, but would cause a reset of the CPU complex
- Nmi would likely just trigger an NMI immediately; no power state change, but affects the state of the host software
- PushPowerButton I think could have different behaviors; I think it would likely turn the system off if it's already on, or turn the system on if it's already off

Reply:
Thanks, Mike and Samer for the comments.

Mike - agree with your comments about the impact of those commands (system changed vs. not changed).

But just to clarify in case it wasn't obvious - the Ansible redfish_command module does not currently have commands for "ForceOn", "ForceRestart", "PushPowerButton", "Nmi", or "PowerCycle". We can add those, but would rather do that as a separate feature issue.

Reply:
Ah, gotcha. Yes, I agree it would make more sense to add those checks when Ansible makes use of new reset types.

Reply:
I agree on making these a separate pull request